src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -1050,8 +1052,19 @@ namespace cryptonote
           break;
         case relay_method::block:
         case relay_method::fluff:
-          fluff_txs.push_back(std::move(tx));
+        {
+          cryptonote::transaction transaction;
+          if(cryptonote::parse_and_validate_tx_from_blob(tx, transaction))
+          {
+            if(fluff_txs_hash.find(transaction.hash) != fluff_txs_hash.end())
+            {
+              break;
+            }
+            fluff_txs_hash.emplace(transaction.hash);
+          }
+          fluff_txs.push_back(std::move(tx));            
           break;
+        }
```

Comment:
I don't think this is needed, right?

We already check for duplicates in the whole message at the top of this function.

Reply:
Aha. Saw the code here: https://github.com/0xFFFC0000/monero/blob/a899386881a79fecae1fbaf0fc463f33367599c6/src/cryptonote_protocol/cryptonote_protocol_handler.inl#L997


We don’t need the code here in `cryptonote_protocol_handler.inl` . 

Thanks for mentioning it.